Add support for the Events entity

dufferzafar opened the pull request:
This is a work in progress. 

I'm just looking at older commits to see what changes need to be made for adding a new entity, and working my way through.

Currently, this is working: 
`m.get_event_by_id("a4a0927c-8ad7-48dd-883c-7126cc0b9c6b", [])`

~~I'll add some tests too.~~

mineo commented:
Related to #154.

dufferzafar commented:
There was a WS bug in browsing events, which has been fixed on beta. [Ian's comment.](http://chatlogs.musicbrainz.org/musicbrainz-devel/2015/2015-06/2015-06-07.html#T18-27-08-245093)

So this works:

```python
m.set_hostname("beta.musicbrainz.org")
print(m.search_events(query="Pink Floyd"))
print(m.browse_events(artist="83d91898-7763-47d7-b03b-b92132375c47", includes=['artist-rels']))

```

alastair commented on test/data/event/770fb0b4-0ad8-4774-9275-099b66627355-place-rels.xml:
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8"?><metadata xmlns="http://musicbrainz.org/ns/mmd-2.0#"><event type="Concert" id="770fb0b4-0ad8-4774-9275-099b66627355"><name>1987-06-07: Rock am Ring, Nürburgring, Nürburg, Germany</name><life-span><begin>1987-06-07</begin><end>1987-06-07</end></life-span><relation-list target-type="place"><relation type-id="e2c6f697-07dc-38b1-be0b-83d740165532" type="held at"> ...
```
Can you format this with newlines (look at other examples in the data directory for number of spaces indent)

JonnyJD replied:
These files only serve for the mocking of the web request and in such should be identical to what the web request returns (which is without newlines).

Of course, for the actual debugging I look at the result of the request in my browser (where it is automatically formatted).

alastair commented on test/test_getentity.py:
```diff
@@ -42,6 +42,11 @@ def testGetArtist(self):
+        event_id = "a4a0927c-8ad7-48dd-883c-7126cc0b9c6b"
+        musicbrainzngs.get_event_by_id(event_id)
+        self.assertEqual("http://musicbrainz.org/ws/2/event/a4a0927c-8ad7-48dd-883c-7126cc0b9c6b", self.opener.get_url())
+
```
I'd like to see some tests here with various includes and release type/status to validate the validation (ha)

alastair commented on test/test_mbxml_event.py:
```diff
@@ -0,0 +1,20 @@
+    def testCorrectId(self):
+        event_id = "770fb0b4-0ad8-4774-9275-099b66627355"
+        res = _common.open_and_parse_test_data(self.datadir, "%s-place-rels.xml" % event_id)
+        self.assertEqual(event_id, res["event"]["id"])
```
Can you check some other stuff in the xml file too?

alastair commented:
This is good, thanks! Just a few small things to fix

alastair commented:
Can you read the Contributing doc to see about adding signed-off-by?

JonnyJD commented:
The new functions need `_autofunction` entries in [docs/api.rst](https://raw.githubusercontent.com/dufferzafar/python-musicbrainzngs/events/docs/api.rst).

Adding an example in [examples/](https://github.com/dufferzafar/python-musicbrainzngs/tree/events/examples) would be nice but is optional for now.
Examples have two goals: 1) showing how to use the API 2) evaluating if the API is easy/intuitive to use.

dufferzafar commented:
@JonnyJD I've removed the newlines from the XML then, and added the autofunction entries but I'm not sure how to document the keys of the event dict.

Something like?
```
    The event dict has the following keys: `id`, `type`, `name`, `time`, `disambiguation`, a `life-span` dict with `begin` and `end` keys.
```

JonnyJD commented:
Thanks for updating the api docs.

```
The event dict has the following keys: 
`id`, `type`, `name`, `time`, `disambiguation` and `life-span`.
```
sounds good. Listing the "first level" should be enough. Otherwise we start duplicating this info.
It is also fairly easy to print the `life-span` dict to find out what's in there compared to printing the whole event just to find out which keys are included.
So this short list should help a lot.

The commits are also signed-off, I also tested this a bit myself, so from my end this is complete. :shipit:
I'll wait if @alastair has anything to add though.

You can add yourself to the list of authors in [README.rst](https://github.com/alastair/python-musicbrainzngs/blob/master/README.rst#authors) and then change in [CHANGES](https://github.com/alastair/python-musicbrainzngs/blob/master/CHANGES) so we don't forget that.

The server-side fixes for browsing events are in [59f380bb0](https://bitbucket.org/metabrainz/musicbrainz-server/commits/59f380bb03a5e414afa8f6d5c52acfbf66dc6486) and [0bf34d687](https://bitbucket.org/metabrainz/musicbrainz-server/commits/0bf34d687e3a5456cd36dc6c8ff4ce9208a9cc21) and will be released on Monday. We don't have to wait for that, though, since events are a new feature for us anyways so nothing breaks.

JonnyJD commented:
:+1:

alastair commented:
Cool, lots of changes coming in these days. I'll try and get everything
reviewd and merged this week.

On Mon, Jun 15, 2015 at 4:43 PM, Johannes Dewender <notifications@github.com
> wrote:

> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alastair/python-musicbrainzngs/pull/168#issuecomment-112093902>
> .
>

JonnyJD commented:
FYI: the only conflict with master is in CHANGES and this is easily solved. This can be solved with a rebase of this branch or during the merge of this branch later.

dufferzafar commented:
I've rebased the branch :)
